### PR TITLE
Handle sms client exceptions

### PR DESCRIPTION
**Pull request by minglis**
Improve error handling in the SMS clients.

- bad json
- bad responses
- unified exceptions

### CODE REVIEW

File: app/clients/sms/mmg.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import json
 from monotonic import monotonic
 from requests import (request, RequestException, HTTPError)
+
 from app.clients import (STATISTICS_DELIVERED, STATISTICS_FAILURE)
-from app.clients.sms import (SmsClient, SmsClientException)
+from app.clients.sms import (SmsClient, SmsClientException, SmsClientResponseException)
```
Comment (servingUpAces) [inline app/clients/sms/mmg.py]: Some unused imports

**Comment by servingUpAces** [comment]
👍

File: app/clients/sms/__init__.py [hunk 2]
```diff
@@ -8,6 +8,18 @@ class SmsClientException(ClientException):
     pass
```
Comment (servingUpAces) [inline app/clients/sms/__init__.py]: The SmsClientException is not used anymore (except in unused imports)

**Comment by minglis** [comment]
Closed - doing this as part of some other rework

**Pull request closed without merging**